src/manager/forms.py

```diff
@@ -173,71 +173,81 @@ def render_choices(choices):
     return [(choice.capitalize(), choice) for choice in c_split]
 
 
+def build_generated_form(form, relations, required_form=None, review_form=None):
+    """ Build form from given user-generated fields. """
+
+    for relation in relations:
+        # Ensure ASCII field names.
+        field_name = relation.element.name.encode('ascii', 'xmlcharrefreplace')
```

Comment:
You're doing this over and over throughout this process. Wouldn't it be useful to create a single method that does this, limiting the possibility of getting `.encode('ascii', 'xmlcharrefreplace')` wrong with any of the points it's used?